storage/rocksdb/rdb_comparator.h

```diff
@@ -82,18 +69,17 @@ class Rdb_rev_comparator : public rocksdb::Comparator {
   Rdb_rev_comparator &operator=(const Rdb_rev_comparator &) = delete;
   Rdb_rev_comparator() = default;
 
-  static int bytewise_compare(const rocksdb::Slice &a,
-                              const rocksdb::Slice &b) {
-    return -Rdb_pk_comparator::bytewise_compare(a, b);
-  }
-
   int Compare(const rocksdb::Slice &a, const rocksdb::Slice &b) const override {
-    return -Rdb_pk_comparator::bytewise_compare(a, b);
+    return rocksdb::ReverseBytewiseComparator()->Compare(a, b);
   }
   const char *Name() const override { return "rev:RocksDB_SE_v3.10"; }
   void FindShortestSeparator(std::string *start,
-                             const rocksdb::Slice &limit) const override {}
-  void FindShortSuccessor(std::string *key) const override {}
+                             const rocksdb::Slice &limit) const override {
+    rocksdb::BytewiseComparator()->FindShortestSeparator(start, limit);
+  }
+  void FindShortSuccessor(std::string *key) const override {
+    rocksdb::BytewiseComparator()->FindShortSuccessor(key);
+  }
```

Comment:
These key shorting logic is OK to me.

Reply:
I think RocksDB's ReverseBytewiseComparatorImpl::FindShortestSeparator() now has a bug and it doesn't shorten key for reverse comparator at all. Let me confirm that.

Reply:
Yes I can confirm that with a simple unit test:
```
TEST_F(ComparatorDBTest, FindShortestSeparator) {
  std::string s1 = "abc1xyz";
  std::string s2 = "abc3xy";

  BytewiseComparator()->FindShortestSeparator(&s1, s2);
  ASSERT_EQ("abc2", s1);

  s1 = "abc5xyztt";

  ReverseBytewiseComparator()->FindShortestSeparator(&s1, s2);
  ASSERT_EQ("abc4", s1);
}
```

The output:
```
[ RUN      ] ComparatorDBTest.FindShortestSeparator
db/comparator_db_test.cc:446: Failure
Value of: s1
  Actual: "abc5xyztt"
Expected: "abc4"

```

Reply:
https://github.com/facebook/rocksdb/pull/3836